### PR TITLE
Build Foundation with DEPLOYMENT_RUNTIME_SWIFT disabled

### PR DESCRIPTION
**Pull request by mario-vera**
Expand Linux platforms supported under DEPLOYMENT_RUNTIME_SWIFT=0 umbrella.

### CODE REVIEW

File: CoreFoundation/Parsing.subproj/CFXMLInterface.c [hunk 1]
```diff
@@ -22,7 +23,7 @@
 #include <libxml/xpath.h>
 #include <libxml/xpathInternals.h>
 #include <libxml/dict.h>
-#include "CFInternal.h"
```
Comment (parkera) [inline CoreFoundation/Parsing.subproj/CFXMLInterface.c]: So we don't use CFInternal.h here at all?

File: CoreFoundation/Parsing.subproj/CFXMLInterface.c [hunk 2]
```diff
@@ -168,6 +176,9 @@ _CFXMLInterfaceSAXHandler _CFXMLInterfaceCreateSAXHandler() {
     saxHandler->externalSubset = (externalSubsetSAXFunc)__CFSwiftBridge.NSXMLParser.externalSubset;
     
     saxHandler->initialized = XML_SAX2_MAGIC; // make sure start/endElementNS are used
+#else
+    memset(&saxHandler, 0, sizeof(saxHandler));
```
Comment (parkera) [inline CoreFoundation/Parsing.subproj/CFXMLInterface.c]: Doesn't the calloc above already promise us zeroed memory?

File: lib/script.py [hunk 3]
```diff
@@ -226,7 +226,15 @@ def generate_products(self):
...
-            script += product.generate()
+            items = product.generate()
+            for item in items:
+                if isinstance(item, list):
+                    for subitem in item:
+                        #TODO: What to do with this elements?
+                        #script += subitem
+                        continue
+                else:
+                    script += item
```
Comment (parkera) [inline lib/script.py]: Can you explain a bit more about what this is doing?

File: CoreFoundation/Parsing.subproj/CFXMLInterface.c (hunk 1 above)
Comment (mario-vera) [reply CoreFoundation/Parsing.subproj/CFXMLInterface.c]: It wasn't removed, it moved up & uses qualified path. Check line 15: <CoreFoundation/CFInternal.h>

File: CoreFoundation/Parsing.subproj/CFXMLInterface.c (hunk 2 above)
Comment (mario-vera) [reply CoreFoundation/Parsing.subproj/CFXMLInterface.c]: Agreed, this line is not needed, I'll remove it.

File: lib/script.py (hunk 3 above)
Comment (mario-vera) [reply lib/script.py]: Please add Lily to review this one. script is a string. items is a list containing strings and lists. On python 2.7 we can not append a list into a string, so we need to walk it, appending the strings on the array. 
The comment is to highlight the fact that I found no usage for the  nested lists... Not sure what the intended usage was.

File: CoreFoundation/Parsing.subproj/CFXMLInterface.c (hunk 1 above)
Comment (parkera) [reply CoreFoundation/Parsing.subproj/CFXMLInterface.c]: Got it, thanks. Misread the diff.

File: lib/script.py (hunk 3 above)
Comment (parkera) [reply lib/script.py]: cc @millenomi

Comment (millenomi) [reply lib/script.py]: IIRC: This path is triggered by the `+= [ … ]` command that adds the block runtime, which is invoked if libdispatch is not available. Can you just build libdispatch as part of your CF build instead?

Comment (millenomi) [reply lib/script.py]: You can also fix that command to add single items instead of a list.

Comment (millenomi) [reply lib/script.py]: Sorry, for context: this is the portion of build.py that is triggering that behavior:

```python
# This code is already in libdispatch so is only needed if libdispatch is
# NOT being used
if "LIBDISPATCH_SOURCE_DIR" not in Configuration.current.variables:
    sources += (['closure/data.c', 'closure/runtime.c'])
```

It'd be nice to fix this bit, but in general you want to set your `LIBDISPATCH_SOURCE_DIR` environment variable correctly.

Comment (mario-vera) [reply lib/script.py]: Applying LIBDISPATCH_SOURCE_DIR did not work for me. I just auggested a more pythonic way of applying the product commands... hopefully that is better.

Comment (mario-vera) [reply lib/script.py]: BTW, I'm compiling from within CoreFoundation, so the build.py I'm using is in CoreFoundation/build.py, that scripts do not have any references to LIBDISPATCH_SOURCE_DIR.

**Comment by millenomi** [comment]
@swift-ci please test

**Comment by millenomi** [comment]
@swift-ci please test and merge

**Pull request merged**